crates/cranelift/src/func_environ.rs

```diff
@@ -3421,6 +3421,20 @@ impl FuncEnvironment<'_> {
     pub fn clif_instruction_traps_enabled(&self) -> bool {
         self.tunables.signals_based_traps || self.isa.triple().is_pulley()
     }
+
+    /// Returns whether loads from the null address are allowed as signals of
+    /// whether to trap or not.
+    pub fn load_from_zero_allowed(&self) -> bool {
+        // Pulley allows loads-from-zero and otherwise this is only allowed with
+        // traps + spectre mitigations.
+        self.isa.triple().is_pulley()
```

Comment:
Might as well use the helper if you're gonna define it:

```suggestion
        self.is_pulley()
```